src/main/java/io/axoniq/axonserver/connector/ResultStreamPublisher.java

```diff
@@ -0,0 +1,145 @@
+        private final AtomicBoolean signalGate = new AtomicBoolean(false);
+        private final AtomicBoolean cancelled = new AtomicBoolean(false);
+        private final AtomicBoolean subscribed = new AtomicBoolean(false);
+        private final AtomicBoolean completed = new AtomicBoolean(false);
+
+        private ResultStreamSubscription(Subscriber<? super Message> subscriber,
+                                         ResultStream<Message> resultStream) {
+            this.subscriber = subscriber;
+            this.resultStream = resultStream;
+            this.resultStream.onAvailable(this::signal);
+        }
+
+        @Override
+        public void request(long n) {
+            if (n <= 0) {
+                subscriber.onError(new IllegalArgumentException("negative subscription request"));
+                return;
+            }
+            requested.updateAndGet(current -> current + Math.min(Long.MAX_VALUE - current, n));
+            signal();
+        }
+
+        @Override
+        public void cancel() {
+            logger.debug("The call has been cancelled.");
+            cancelled.set(true);
+            resultStream.close();
+        }
+
+        private void signal() {
+            while (canSignal() && signalGate.compareAndSet(false, true)) {
+                try {
+
+                    long requests = requested.get();
+                    long counter = 0;
+                    for (int i = 0; i < requests; i++) {
+                        if (!resultStream.isClosed() && resultStream.peek() != null) {
+                            subscriber.onNext(resultStream.next());
+                            counter--;
+                        } else {
+                            break;
+                        }
+                    }
+                    requested.getAndAccumulate(counter, Long::sum);
+
+                    if (resultStream.isClosed()) {
+                        subscriber.onComplete();
+                        completed.set(true);
+                    }
+                    resultStream.getError().ifPresent(this::onError);
+                } catch (InterruptedException e) {
+                    signalGate.set(false);
+                    Thread.currentThread().interrupt();
+                } catch (Exception ex) {
+                    this.onError(ex);
+                } finally {
+                    signalGate.set(false);
+                }
+            }
+        }
+
+        private void onError(Throwable error) {
+            logger.debug("An error occurred accessing the ResultStream.", error);
+            subscriber.onError(error);
+            completed.set(true);
+        }
+
+
+        private boolean canSignal() {
+            try {
+                return subscribed.get() &&
+                        !cancelled.get() &&
+                        !completed.get() &&
+                        (resultStream.isClosed() || (resultStream.peek() != null && requested.get() > 0));
```

Comment:
Super Nit: I believe the per-line separated approach would put the operators at the start. Or, that's how it's done throughout the rest of this project
So, something like this:

```suggestion
                return subscribed.get() 
                    && !cancelled.get() 
                    && !completed.get() 
                    && (resultStream.isClosed() || (resultStream.peek() != null && requested.get() > 0));
```

Reply:
Done